seed/challenges/02-javascript-algorithms-and-data-structures/es6.json

```diff
@@ -71,66 +67,29 @@
-        "Instructions",
-        "Fix the code so each camper shows their correct index when they are called.",
-        "Note",
-        "Remember to add \"use strict\" to the top of your code."
+        "<blockquote>var numArray = [];<br>for (var i = 0; i < 3; i++) {<br>  numArray.push(i);<br>}<br>console.log(numArray);<br>// returns [0, 1, 2]<br>console.log(i);<br>// returns 3</blockquote>",
+        "With the <code>var</code> keyword, <code>i</code> is declared globally. So when <code>i++</code> is executed, it updates the global variable. This code is similiar to the following:",
+        "<blockquote>var numArray = [];<br>var i;<br>for (i = 0; i < 3; i++) {<br>  numArray.push(i);<br>}<br>console.log(numArray);<br>// returns [0, 1, 2]<br>console.log(i);<br>// returns 3</blockquote>",
+        "This behavior will cause problems when you create a function and store it for later use inside the for loop that uses the <code>i</code> variable. This is because the stored function will always refer to the value of the updated global <code>i</code> variable.",
+        "<blockquote>var printNumTwo;<br>for (var i = 0; i < 3; i++) {<br>  if(i === 2){<br>      printNumTwo = function() {<br>          return i;<br>      };<br>  }<br>}<br>console.log(printNumTwo());<br>// returns 3</blockquote>",
+        "As you can see, <code>printNumTwo()</code> prints 3 and not 2. This is because the value assigned to <code>i</code> was updated and the <code>printNumTwo()</code> returns the global <code>i</code> and not the value <code>i</code> had when the function was created in the for loop. The <code>let</code> keyword does not follow this behavior:",
+        "<blockquote>'use strict';<br>let printNumTwo;<br>for (let i = 0; i < 3; i++) {<br>  if (i === 2) {<br>      printNumTwo = function() {<br>          return i;<br>     };<br>  }<br>}<br>console.log(printNumTwo());<br>// returns 2<br>console.log(i);<br>// returns \"i is not defined\"</blockquote>",
+        "<code>i</code> is not defined because it was not declared in the global scope. It is only declared within the for loop statement. <code>printNumTwo()</code> returned the correct value because three different <code>i</code> variables with unique values (0, 1, and 2) were created by the <code>let</code> keyword within the loop statement.",
+        "<hr>",
+        "Fix the code so that each camper returns their correct index when they are called.",
+        "<strong>Note</strong><br>Remember to add \"use strict\" to the top of your code."
       ],
       "challengeSeed": [
         "var newCampers = [{camper: \"Wil\"}, {camper: \"Sam\"},{camper: \"Dav\"}];",
         "// only change code below this line",
-        "for( var i = 0; i < newCampers.length; i++){",
+        "for (var i = 0; i < newCampers.length; i++) {",
         "// only change code above this line",
-        "    newCampers[i].roleCall = function(){",
+        "    newCampers[i].roleCall = function() {",
```

Comment:
@BrendanSweeny Where should we add that `console.log(typeof i);` in that case? Inside the the `roleCall` function? To the end of the `challengeSeed`? :blush:

@freeCodeCamp/moderators What do you think about this example?